src/components/pages/ProfilePage.js

```diff
@@ -1,12 +1,54 @@
-import { React } from "react";
+import { React, useContext, useEffect, useState } from "react";
 import { useAuthenticator } from "@aws-amplify/ui-react";
 import { FaArrowRight } from "react-icons/fa";
 import Navbar from "../reusable/NavBar";
 import Sidebar from "../reusable/SideBar";
+import AlgoquantApiContext from "../../ApiContext";
+import LoadSpinner from "../reusable/LoadSpinner";
 
 const ProfilePage = () => {
   const { user } = useAuthenticator((context) => [context.user]);
   const { signOut } = useAuthenticator((context) => [context.user]);
+  const algoquantApi = useContext(AlgoquantApiContext);
+  const [jwtToken, setJwtToken] = useState(null);
+
+  const [balance, setBalance] = useState();
+  const [isLoading, setIsLoading] = useState(true);
+  const [alpacaConnection, setAlpacaConnection] = useState(false);
+
+  // setJwtToken(user?.signInUserSession?.accessToken?.jwtToken);
+  useEffect(() => {
+    const getToken = async () => {
+      try {
+        setJwtToken(user?.signInUserSession?.accessToken?.jwtToken);
+      } catch (error) {
+        console.log(error);
+        setJwtToken(null);
+      }
+    };
+    getToken();
+  }, [user]);
+
+  useEffect(() => {
+    if (jwtToken) {
+      algoquantApi.getUser(jwtToken).then((resp) => {
+        console.log(resp);
+        setBalance(resp.data.Item.buying_power);
+        console.log(resp.data.Item);
+        if (
+          typeof resp.data.Item.alpaca_secret_key !== "undefined" ||
```

Comment:
We will not be returning the Alapca keys , we will just be returning a boolean that represents if the user is using Alapca or not. This should be checking that boolean not the keys themselves .

Reply:
fixed to use the new field from response